src/main/java/org/mongojack/JacksonDBCollection.java

```diff
@@ -2156,11 +2145,26 @@ DBObject convertToBasicDbObject(T object) throws MongoException {
      * @return a mongo DBObject serialized with the ObjectMapper for this collection.
      * @throws MongoException
      */
-    public DBObject convertToDbObject(T object) throws MongoException {
```

Comment:
It would be nice to leave this method as is. In your new version `isEnabled(Feature.USE_STREAM_SERIALIZATION)` is being copy/pasted many times whereas this method cuts down on code duplication

I'm fine with you adding the new method below though, which seems like it would solve much of your issue:

    public static <T> DBObject convertToDbObject(T object, boolean useStreamSerialization, Class<?> view, ObjectMapper objectMapper)